lms/djangoapps/program_enrollments/link_program_enrollments.py

```diff
@@ -0,0 +1,184 @@
+""" Management command to link program enrollments and external student_keys to an LMS user """
+import logging
+
+from django.contrib.auth import get_user_model
+from django.db import IntegrityError, transaction
+from lms.djangoapps.program_enrollments.models import ProgramEnrollment
+from student.models import CourseEnrollmentException
+
+logger = logging.getLogger(__name__)
+User = get_user_model()
+
+NO_PROGRAM_ENROLLMENT_TPL = (u'No program enrollment found for program uuid={program_uuid} and external student '
+                             'key={external_student_key}')
+NO_LMS_USER_TPL = u'No user found with username {}'
+COURSE_ENROLLMENT_ERR_TPL = u'Failed to enroll user {user} with waiting program course enrollment for course {course}'
+EXISTING_USER_TPL = (u'Program enrollment with external_student_key={external_student_key} is already linked to '
+                     u'{account_relation} account username={username}')
+
+
+@transaction.atomic
```

Comment:
Is there anything that writes data to the DB in this function that isn't `link_program_enrollment_to_lms_user()`?  If not, we may not need the atomic decorator on the function.

Reply:
I don't think it hurts, and in case anything blows up it's maybe a good idea to have it on there?